packages/react-liveness/src/components/FaceLivenessDetector/service/utils/blazefaceFaceDetection.ts

```diff
@@ -67,33 +66,33 @@ export class BlazeFaceFaceDetection extends FaceDetection {
 
     const timestampMs = Date.now();
 
-    const faces: Face[] = predictions.map((prediction) => {
-      const { topLeft, bottomRight, probability, landmarks } = prediction;
-
-      if (landmarks === undefined) return;
-
-      const [right, top] = topLeft as [number, number]; // right, top because the prediction is flipped
-      const [left, bottom] = bottomRight as [number, number]; // left, bottom because the prediction is flipped
-      const width = Math.abs(right - left);
-      const height = Math.abs(bottom - top);
-      const rightEye = landmarks[0] as Face['rightEye'];
-      const leftEye = landmarks[1] as Face['leftEye'];
-      const nose = landmarks[2] as Face['nose'];
-      const mouth = landmarks[3] as Face['mouth'];
-
-      return {
-        top,
-        left,
-        width,
-        height,
-        timestampMs,
-        probability: probability[0] as number,
-        rightEye,
-        leftEye,
-        mouth,
-        nose,
-      };
-    });
+    const faces: Face[] = predictions
+      .filter((prediction) => !!prediction.landmarks)
+      .map((prediction) => {
+        const { topLeft, bottomRight, probability, landmarks } = prediction;
+
+        const [right, top] = topLeft as [number, number]; // right, top because the prediction is flipped
+        const [left, bottom] = bottomRight as [number, number]; // left, bottom because the prediction is flipped
+        const width = Math.abs(right - left);
+        const height = Math.abs(bottom - top);
+        const rightEye = (landmarks as number[][])[0];
+        const leftEye = (landmarks as number[][])[1];
+        const nose = (landmarks as number[][])[2];
+        const mouth = (landmarks as number[][])[3];
```

Comment:
Obviously it would be ideal if the `NormalizedFace` interface would allow us to not have to cast these values, but  get that we cannot because of the type unions in the interface. Since we are not using type predicates here what do you think of adding some meaningfully named utility types for `[number, number]` and `number[][]`?

Reply:
Yeah I think that makes sense. I think that the types that would make sense here are probably something like:

`type Coordinate = [number, number]`

and `landmarks` being something like `Coordinate[]`

what do you think?

Reply:
Makes sense to me!